packages/react/src/components/Geo/LocationSearch/index.tsx

```diff
@@ -33,20 +60,37 @@ const LocationSearchControl = ({
   return null;
 };
 
-const LocationSearchStandalone = (props: LocationSearchProps) => {
-  const hasMounted = useRef(false);
+const LocationSearchStandalone = ({
+  onLoading,
+  onResult,
+  onResults,
+  onClear,
+  ...props
+}: LocationSearchProps) => {
+  const geocoderRef = useRef(null);
 
   useEffect(() => {
-    if (!hasMounted.current) {
-      (createAmplifyGeocoder(props) as unknown as AmplifyLocationSearch).addTo(
-        `#${LOCATION_SEARCH_CONTAINER}`
-      );
+    const map = createAmplifyGeocoder(
+      props
+    ) as unknown as AmplifyLocationSearch;
+    geocoderRef.current = map;
 
-      hasMounted.current = true;
-    }
-  }, [props]);
+    map.addTo(`#${LOCATION_SEARCH_CONTAINER}`);
+
+    map.on('result', onResult);
+    map.on('loading', onLoading);
+    map.on('results', onResults);
+    map.on('clear', onClear);
+
+    return () => {
+      map.off('result', onResult);
+      map.off('loading', onLoading);
+      map.off('results', onResults);
+      map.off('clear', onClear);
+    };
+  }, [onResult, onLoading, onResults, onClear, props]);
```

Comment:
Just want to confirm, are there any issues with this `useEffect` running on every render? For context, `props` will have a new reference on every render since it is created from the spread operator on line 68